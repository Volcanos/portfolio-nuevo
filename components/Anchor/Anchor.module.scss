.anchor {
  font-size: 1rem;
  text-decoration: none;
  position: relative;
  &:hover {
    color: #6f6f6f;
    cursor: pointer;

    &:before,
    &:after {
      width: 50%;
      opacity: 1;
    }
  }

  &:before,
  &:after {
    content: "";
    position: absolute;
    bottom: 0px;
    width: 0px;
    height: 1px;
    margin: 2px 0 0;
    transition: all 0.2s ease-in-out;
    transition-duration: 0.65s;
    opacity: 0;
    background-color: #252525;
  }
  &:before {
    left: 50%;
  }
  &:after {
    right: 50%;
  }
}
