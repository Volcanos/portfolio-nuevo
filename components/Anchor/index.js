import Link from 'next/link';
import styles from './Anchor.module.scss';

const Anchor = (props) => {
    let prefetch = typeof props.prefetch !== 'undefined' ? props.prefetch : true;
    let properties = prefetch ? {href: props.url, as: props.as} : {href: props.url, as: props.as, prefetch: false};
    return (
        <Link {...properties}>
            <span className={styles.anchor} style={props.style}>
                {props.children}
            </span>
        </Link>
    )
}
export default Anchor;
