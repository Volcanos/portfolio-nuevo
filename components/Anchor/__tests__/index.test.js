import React from 'react'
import { mount, render } from 'enzyme';
import 'jest-styled-components'
import Anchor from '../index';

describe('Anchor', function () {
    it('should render to static and contain Enlace', function () {
        const text = render(<Anchor url="/">Enlace</Anchor>).text();
        expect(text).toEqual('Enlace');
    });

    it('Basic render snapshot not change', function () {
        const tree = render(<Anchor url="/">Enlace</Anchor>);
        expect(tree).toMatchSnapshot();
    });
    it('Complete render snapshot not change', function () {
        const tree = render(<Anchor url="/pid" as="/post-con-titulo" style={{color: '#fff'}} fontSize="16px">Enlace</Anchor>);
        expect(tree).toMatchSnapshot();
    });
});