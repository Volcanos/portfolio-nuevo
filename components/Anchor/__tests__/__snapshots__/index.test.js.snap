// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Basic render snapshot not change 1`] = `
.c0 {
  font-size: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
}

.c0:hover {
  color: #6f6f6f;
  cursor: pointer;
}

.c0:hover:before,
.c0:hover:after {
  width: 50%;
  opacity: 1;
}

.c0:before,
.c0:after {
  content: "";
  position: absolute;
  bottom: 0px;
  width: 0px;
  height: 1px;
  margin: 2px 0 0;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-transition-duration: 0.65s;
  transition-duration: 0.65s;
  opacity: 0;
  background-color: #6f6f6f;
}

.c0:before {
  left: 50%;
}

.c0:after {
  right: 50%;
}

<a
  class="c0"
>
  Enlace
</a>
`;

exports[`Anchor Complete render snapshot not change 1`] = `
.c0 {
  font-size: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
}

.c0:hover {
  color: #6f6f6f;
  cursor: pointer;
}

.c0:hover:before,
.c0:hover:after {
  width: 50%;
  opacity: 1;
}

.c0:before,
.c0:after {
  content: "";
  position: absolute;
  bottom: 0px;
  width: 0px;
  height: 1px;
  margin: 2px 0 0;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-transition-duration: 0.65s;
  transition-duration: 0.65s;
  opacity: 0;
  background-color: #6f6f6f;
}

.c0:before {
  left: 50%;
}

.c0:after {
  right: 50%;
}

<a
  class="c0"
  style="color:#fff"
>
  Enlace
</a>
`;
